open-craft/opencraft
# BB-1837: Start/stop services when starting/stopping vms.

This PR implements methods to start/stop services on VMs that are being started/stopped.

**Testing instructions:**
1. Checkout this branch on Stage.
2. Pick an instance with two running appservers (such as [this one](https://stage.console.opencraft.com/instance/4210/).
3. Activate one appserver, check on the logs that the playbook to enable services was correctly run.
4. Deactivate one appserver, check tat the appserver was correctly run, SSH into the VM and check that the services were stopped.
5. Activate the same appserver, and check that the services were started.
6. Rename `/edx/bin/supervisord` to something else, and deactivate the appserver, it should work.
7. Try to activate the broken appserver, the appserver should not be activated.
8. Repeat this a few times.


**Reviewers:**

CC @pomegranited and @toxinu
